src/platform/mbed/arch.c

```diff
@@ -48,4 +49,9 @@ uint64_t __atomic_exchange_8(volatile void* ptr, uint64_t val, int memorder) {
 bool __atomic_compare_exchange_8(volatile void *ptr, void*expected, unsigned long long desired, int success_memorder, int failure_memorder)
 {
     return core_util_atomic_cas_explicit_u64((volatile uint64_t*)ptr, (uint64_t*)expected, desired, mem_order(success_memorder), mem_order(failure_memorder));
+}
+
+void usleep(unsigned int usec) 
+{
+    wait_us(usec);
```

Comment:
I think this could be divided between `ThisThread::sleep_for` for milliseconds and `wait_us` for us: 

```c++
unsigned int ms = (usec / 1000);
unsigned int us = usec % 1000;

if (ms) {
    ThisThread::sleep_for(ms);
}

if (us) { 
    wait_us(us);
}
```

Reply:
Done.
BTW. I had to replace rtos::ThisThread::sleep_for() with its 'C' version: thread_sleep_for() [platform/mbed_thread.h].